### PR TITLE
Add getAnnotations function to plugin and remove _getState private one

### PR DESCRIPTION
**Pull request by stockiNail**
This PR is adding `getAnnotations` function to annotation plugin to get all annotation elements created for a chart.
It also removes  _getState` function from annotation plugin because it was used for testing and it was the only way to access to the elements instances.

### TODO

- [x] test cases
- [x] types
- [x] documentation

### CODE REVIEW

**Comment by davidnir1** [comment]
Hello @stockiNail, 
Is there an ETA on this PR (if approved)?
We're trying to prioritize tasks and this solves a big one for us :)
Thanks!

**Comment by tomasbeloch** [comment]
Hello @stockiNail, is there some plan in place when you're planing to merge this one? It would be great to allow the tooltips over the box annotations. Thanks!

**Comment by stockiNail** [comment]
@tomasbeloch I don't think so, I'm a collaborator not a maintainer ;) Furthermore I'm currently busy in another important project therefore my time is very limited. I'm very sorry. I hope to have time soon to restart working on this project.
@kurkle what do you think?

File: src/events.js [hunk 1]
```diff
@@ -96,7 +95,7 @@ function dispatchMoveEvents({state, event}, hook, elements, checkElements) {
 
 function handleClickEvents(state, event, options) {
   const listeners = state.listeners;
-  const elements = getElements(state, event, options.interaction);
+  const elements = getElements(state.visibleElements, event, options.interaction);
```
Comment (LeeLenaleee) [inline src/events.js:98]: isn't this a breaking change if we only use visible elements now while normally we used all elements?

Comment (stockiNail) [reply src/events.js:98]: No, it isn't.

the `state.visibleElements` was invoked previously in the function mode (see changes below):

```javascript
function filterElements(state, event, options) {
  return state.visibleElements.filter
```

**Review by LeeLenaleee — APPROVED** [review]

**Review by LeeLenaleee — APPROVED** [review]

**Pull request merged**